src/hexo/themes/documentation/layout/partials/docsearch.hbs

```diff
@@ -1,10 +1,28 @@
 <script src="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js"></script>
 <script>
-    docsearch({
-        apiKey: '2ee58f908e6d720c4f72a9645ce0e857', // search-only API key, okay for frontend
-        indexName: 'sonarwhal',
-        inputSelector: '#search-input',
-        appId: '0R8E5SPAFP',
-        debug: false // set to true if need to debug css
-    });
+    (function () {
+        var search;
+        var previewSize = 500;
+        var transformInput = function () {
+            if (window.innerWidth > previewSize) {
+                // transform the input field to an algolia search input if there is enough space
+                search = docsearch({
+                    apiKey: '2ee58f908e6d720c4f72a9645ce0e857', // search-only API key, okay for frontend
+                    indexName: 'sonarwhal',
+                    inputSelector: '#search-input',
+                    appId: '0R8E5SPAFP',
+                    debug: false, // set to true if need to debug css
+                });
+            } else {
+                // transform the input field to a normal input field
+                // hitting enter will redirect to the search result page
+                if (search) {
+                    search.autocomplete.autocomplete.destroy();
+                }
+            }
+        };
+
+        window.addEventListener('resize', transformInput);
```

Comment:
There are a couple things that can be improved in here:

1.  `resize` events can be fired at a high rate so we should throttle them. There are some examples [here](https://developer.mozilla.org/en-US/docs/Web/Events/resize). Because we don't have to update the layout here you can maybe use a timer and set it to 200ms or so. In EdgePortal we have some examples too for the search in the different sections.
1. We create and destroy the `search` object each time. Maybe you can try replacing `open` and see if it doesn't show up? Something like the following:

```js
var search = docsearch({...});
var originalOpen = search.open;
var fakeOpen= function() {};

var handleResults = function () {
  if (window.innerWidth > previewSize) { 
    search.open = originalOpen;
  } else {
    search.open = fakeOpen;
  }
}
```